OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -132,8 +136,38 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
 		}
 
+		void UpdateTarget()
+		{
+			if (args.GuidedTarget.IsValidFor(args.SourceActor))
+			{
+				var guidedTargetPos = args.GuidedTarget.CenterPosition;
+				var targetDistance = new WDist((guidedTargetPos - args.Source).Length);
+
+				// Only continue tracking target if it's within weapon range +
+				// BeyondTargetRange to avoid edge case stuttering (start firing and immediately stop again).
+				if (targetDistance > args.Weapon.Range + info.BeyondTargetRange)
+				{
+					continueTracking = false;
+					isTailTravelling = true;
```

Comment:
Can you please pull out a `StopTargeting` method that is used here and when the target dies?  This will cut down on the number of places that directly manipulate these flags.